### PR TITLE
`Exam mode`: Improve performance when navigating into exam with many programming exercises

### PR DESCRIPTION
**Pull request by krusche**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context
We load too much data, therefore navigating into some pages in exams can become rather slow

### Description
This PR reduces the amount of data which is loaded.

### Steps for Testing
- Test that the exercise group overview page still shown all relevant information (for PE the template and solution repository link, the amount of quiz questions for quiz exercises)
- Test that the other exam management pages stil work correctly.

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by dearjasmina — DISMISSED** [review]
code lgtm

File: src/main/java/de/tum/in/www1/artemis/repository/ExamRepository.java [hunk 1]
```diff
@@ -405,6 +421,12 @@ default Exam findWithExerciseGroupsAndExercisesByIdOrElseThrow(long examId) thro
         return findWithExerciseGroupsAndExercisesById(examId).orElseThrow(() -> new EntityNotFoundException("Exam", examId));
     }
 
+    @NotNull
+    default Exam findWithExerciseGroupsAndExercisesAndDetailsByIdOrElseThrow(long examId) {
+        return findWithExerciseGroupsAndExercisesAndDetailsById(examId).orElseThrow(() -> new EntityNotFoundException("Exam", examId));
+
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/repository/ExamRepository.java:427]:
```suggestion
```

**Review by rriyaldhi — APPROVED** [review]
Tested with 10 Programming Exercises [here](https://artemis-test3.artemis.cit.tum.de/course-management/25/exams/22/) and all looks good.

**Comment by Kroko-fant** [comment]
Tested on ts3 in a testing session worked as expected

**Review by Kroko-fant — APPROVED** [review]

**Review by basak-akan — APPROVED** [review]
Tested during the testing session exercise group overview page worked as described and no issues found in other exam management pages.

**Review by sarpsahinalp — APPROVED** [review]
Tested on ts3 in a testing session worked as expected

**Pull request merged**